cmd/kubemark/hollow-node.go

```diff
@@ -186,7 +186,7 @@ func run(config *hollowNodeConfig) {
 			klog.Fatalf("Failed to create a client config: %v. Exiting.", err)
 		}
 
-		clientFromConfig, err := clientset.NewForConfig(cfg)
+		clientFromConfig, err := clientset.NewForConfig(restclient.AddUserAgent(cfg, "hollow-node"))
```

Comment:
I remember user-agent for hollow node is kubemark.

Reply:
they are all with the kubecfg: for example:

/api/v1/tenants/monkey/namespaces/fy67ad-testns/pods/latency-deployment-4-6b5ccbc79d-qkxbt/status","verb":"patch","user":{"username":"kubecfg"

Reply:
Where is the user agent shown? Only in audit log? As I don't see it in Prometheus data

![request-promethus](https://user-images.githubusercontent.com/7363144/122789129-d57ced80-d26b-11eb-84e2-feb0d27d18aa.PNG)

Reply:
they are in the api server audit logs. you can try to grep one PODID to get all the audit-ed operations on the pod. username is tracked to see who ( which components ) makes the change or requests.